lib/more_core_extensions/core_ext/process/pause_resume.rb

```diff
@@ -0,0 +1,110 @@
+module MoreCoreExtensions
+  module ProcessPauseResume
+    if Gem.win_platform?
+      require 'fiddle'
+      ntdll = Fiddle.dlopen('ntdll')
+      kernel32 = Fiddle.dlopen('kernel32')
+
+      NtSuspendProcess = Fiddle::Function.new(
+        ntdll['NtSuspendProcess'],
+        [Fiddle::TYPE_UINTPTR_T],
+        Fiddle::TYPE_INT
+      )
+
+      private_constant :NtSuspendProcess
+
+      NtResumeProcess = Fiddle::Function.new(
+        ntdll['NtResumeProcess'],
+        [Fiddle::TYPE_UINTPTR_T],
+        Fiddle::TYPE_INT
+      )
+
+      private_constant :NtResumeProcess
+
+      OpenProcess = Fiddle::Function.new(
+        kernel32['OpenProcess'],
+        [Fiddle::TYPE_LONG, Fiddle::TYPE_INT, Fiddle::TYPE_LONG],
+        Fiddle::TYPE_UINTPTR_T
+      )
+
+      private_constant :OpenProcess
+
+      CloseHandle = Fiddle::Function.new(
+        kernel32['CloseHandle'],
+        [Fiddle::TYPE_UINTPTR_T],
+        Fiddle::TYPE_INT
+      )
+
+      private_constant :CloseHandle
+
+      PROCESS_SUSPEND_RESUME = 0x00000800
+
+      private_constant :PROCESS_SUSPEND_RESUME
+    end
+
+    # Returns whether or not the given process is running.
+    #
+    def alive?(pid)
+      Process.kill(0, pid)
```

Comment:
Should the `0` be `"EXIT"` instead for consistency with the other methods?

Reply:
@bdunne Turns out that doesn't work. Courtesy of @agrare:
```
* signal.c (trap): SIGEXIT is not a system signal and is dealt
      with internally, so it should not try to register the system
      signal handler by sigaction.
```
Attempting to use it will result in: `ArgumentError (unsupported name `SIGEXIT')`

Reply:
[`kill(2)`](http://man7.org/linux/man-pages/man2/kill.2.html) says that if sig is 0 then no signal is sent, and there is no `SIGEXIT` defined so :+1: for this from me

Reply:
Interesting, I was looking at https://ruby-doc.org/core-2.7.0/Signal.html#method-c-list

Reply:
Yeah if the intent is for anything in `Signal.list` to be able to be used by `Process.kill` then this looks like a ruby bug